tests/webgl_timer_query.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+#include <emscripten.h>
+#include <emscripten/html5.h>
+
+#define GL_GLEXT_PROTOTYPES
+#ifdef TEST_WEBGL2
+#include <GLES3/gl3.h>
+#include <GLES3/gl2ext.h>
+#else
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+#endif
+
+int result = 0;
+GLuint timerQuery = 0;
+
+#define GL_CALL( x ) \
+    { \
+        x; \
+        GLenum error = glGetError(); \
+        if( error != GL_NO_ERROR ) { \
+            printf( "GL ERROR: %d,  %s\n", (int)error, #x ); \
+            result = 1; \
+        } \
+    } \
+
+void getQueryResult()
+{
+  /* Get the result. It should be nonzero. */
+  GLuint64 time = 0;
+#ifdef TEST_WEBGL2
+  GL_CALL(glGetQueryObjectui64vEXT(timerQuery, GL_QUERY_RESULT, &time));
+#else
+  GL_CALL(glGetQueryObjectui64vEXT(timerQuery, GL_QUERY_RESULT_EXT, &time));
+#endif
+
+  if(!time) return;
+
+  printf("queried time: %llu\n", time);
+  emscripten_cancel_main_loop();
+
+#ifdef TEST_WEBGL2
+  GL_CALL(glDeleteQueries(1, &timerQuery));
+#else
+  GL_CALL(glDeleteQueriesEXT(1, &timerQuery));
```

Comment:
Should either of these work on the other context - i.e. glDeleteQueries when using WebGL 1 or glDeleteQueriesEXT when using WebGL2? I think at least the latter should work, if not both.

Would be nice to test this. We could do this by splitting TEST_CORE_QUERIES from TEST_WEBGL2  and testing 3-4 combinations of them. WDYT?

Reply:
Haha, sorry, I already swapped out all knowledge I had about this... but if my sparse memory serves me right, it's not interchangeable:

- the EXT version works only on WebGL 1 contexts as it comes from the [EXT_disjoint_timer_query](https://www.khronos.org/registry/webgl/extensions/EXT_disjoint_timer_query/) (which, according to the spec, shouldn't even be exposed on WebGL 2)
- the unsuffixed version works only on WebGL 2 contexts as [EXT_disjoint_timer_query_webgl2](https://www.khronos.org/registry/webgl/extensions/EXT_disjoint_timer_query_webgl2/) relies on the core WebGL 2 query functionality for everything except 64-bit query result retrieval

Maybe you're referring to the original version of this PR (from 2019) which had some compatibility bits to make the EXT variants working on WebGL 2 as well (and thus mirroring the ES [EXT_disjoint_timer_query](https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_disjoint_timer_query.txt), which allows the EXT functions on ES3)? I dropped that since, because the maintenance overhead didn't seem to be worth it (and I didn't have any need for it in Magnum either).

Reply:
Got it, I think you're right I got confused with the original version. Just wanted to make sure we're testing whatever is implemented.